### PR TITLE
wikipedia: support media fragments

### LINKED ISSUE #2316: wikipedia: not all fragments are sections
### Description

MediaWiki uses document fragments for view state in some cases, such as when clicking an image thumbnail to open it in the full viewport. Sopel doesn't like this.

### Reproduction steps

0. Ensure that `wikipedia` plugin is loaded
1. Send a link to e.g. https://en.wikipedia.org/wiki/Diffraction_spike#/media/File:Comparison_strut_diffraction_spikes.svg
2. Note error (`Error fetching section "/media/File:Comparison_strut_diffraction_spikes.svg" for page "Diffraction spike".`)

### Expected behavior

Sopel should fetch either the article synopsis (fallback to ignoring fragment), or info about the media.

The third option is ignoring the link entirely, I suppose.

### Relevant logs

_No response_

### Notes

Inspired by #2315, which sort of mitigates a symptom of this (excess highlights) but doesn't treat the cause.

### Sopel version

5c4f56e505442031bea4affaa00cae722b61111d

### Installation method

`pip install`

### Python version

3.9.12

### Operating system

_No response_

### IRCd

_No response_

### Relevant plugins

wikipedia

### PR DESCRIPTION
**Pull request by SnoopJ**
### Description
This PR adds support for Wikipedia links to media embedded in articles (i.e. fragments of the form `#/media/File:…`), which closes #2316

**Edit:** While writing up the follow-up issue, I realized that `File:` is specific to English Wikipedia, so in 63187e9 the prefix check was relaxed to support other references to files on other language communities on Wikipedia. It seems to Just Work™ and the description retrieval is fairly cautious anyway.

Note that this changeset **does not** emit an image description for links directly to a `File:…` article, but it's probably possible to rework it to do that too, if desired

### Checklist
- [x] I have read [CONTRIBUTING.md](https://github.com/sopel-irc/sopel/blob/master/CONTRIBUTING.md)
- [x] I can and do license this contribution under the EFLv2
- [x] No issues are reported by `make qa` (runs `make quality` and `make test`)
- [x] I have tested the functionality of the things this change touches

### CODE REVIEW

**Comment by dgw** [comment]
About handling File links directly: We probably don't want to open that can of worms this late in 8.0. But it can be a feature request for 8.1 if you see people post those. Maybe we should ask @xnaas if it's worthwhile, since SackBot gets a lot of traffic.

The PR itself, though, I mentioned on IRC that the commit messages don't follow project standards. Beyond that, not all files/images on Wikipedia come from Commons. You said it's intended that e.g. https://en.wikipedia.org/wiki/Google#/media/File:Google1998.png will not output anything, but that seems confusing for users if some (but not all) image links output a description.

File: sopel/modules/wikipedia.py [hunk 1]
```diff
@@ -272,18 +272,74 @@ def mw_section(server, query, section):
...
+    desc = mw_image_description(server, image)
+
+    if desc:
+        bot.say(desc, truncation=" […]")
+
+
+def mw_image_description(server, image):
+    """
+    Retrieves the description for the given image
+    """
```
Comment (dgw) [inline sopel/modules/wikipedia.py:285]: Nit, matching style elsewhere in the file:

```suggestion
    """Retrieves the description for the given image."""
```

File: sopel/modules/wikipedia.py [hunk 2]
```diff
@@ -272,18 +272,74 @@ def mw_section(server, query, section):
...
+            page = pages[pageids[0]]
+
+        raw_desc = page["imageinfo"][0]["extmetadata"]["ImageDescription"]["value"]
+
+        parser = WikiParser(image)
+        parser.feed(raw_desc)
+        desc = parser.get_result()
+        desc = ' '.join(desc.split())  # collapse multiple whitespace chars
+
+        return desc
```
Comment (dgw) [inline sopel/modules/wikipedia.py:318]: If these throw any exceptions, they won't be caught (these operations don't raise `LookupError`), so why put them inside the `try`?

File: sopel/modules/wikipedia.py [hunk 3]
```diff
@@ -272,18 +272,74 @@ def mw_section(server, query, section):
...
+    except LookupError:
+        return None
```
Comment (dgw) [inline sopel/modules/wikipedia.py:320]: Gonna throw back something similar to what you said on one of my PRs not too long ago: No logging of what went wrong, even only if debug logging is turned on?

**Review by dgw — COMMENTED** [review]
I haven't gone hard enough at this to formally "Request changes", but I did have a couple comments.

File: sopel/modules/wikipedia.py (hunk 2 above)
Comment (SnoopJ) [reply sopel/modules/wikipedia.py]: Good point, fixed in 9642880

File: sopel/modules/wikipedia.py (hunk 3 above)
Comment (SnoopJ) [reply sopel/modules/wikipedia.py:313]: Oh how the tables have turned…  Fair enough, see 3c9e24b, which actually helped me find a bug that led to 48e807e 😁

**Comment by dgw** [comment]
> ༼ つ ◕_◕ ༽つ 8.0

`pip install git+https://github.com/sopel-irc/sopel.git`

**Review by dgw — COMMENTED** [review]
Nit, but you can squash that linter fixup anyway. 😉

**Review by Exirel — APPROVED** [review]

**Pull request merged**